crates/sui-core/src/consensus_validator.rs

```diff
@@ -62,7 +62,8 @@ impl SuiTxValidator {
                 ConsensusTransactionKind::CertifiedTransaction(certificate) => {
                     cert_batch.push(certificate.as_ref());
                 }
-                ConsensusTransactionKind::CheckpointSignature(signature) => {
+                ConsensusTransactionKind::CheckpointSignature(signature)
+                | ConsensusTransactionKind::CheckpointSignatureV2(signature) => {
```

Comment:
we should reject CheckpointSignatureV2 if it is not enabled according to the protocol config. 

@mwtian can you confirm that behavior will be identical for both the case where a transaction fails to deserialize (if an old validator build sees the new message) vs the case where it is rejected here? (if a new validator sees the message but it is not enabled).

Reply:
Yes, if an error is returned from this function, the block would be rejected, same as for deserialization failures.